File: WinForms/CS/RichEditMainDemo/Modules/CustomDraw/CustomsRichEditCommandFactoryService.cs
using DevExpress.Office;
using DevExpress.XtraEditors;
using DevExpress.XtraRichEdit.Commands;
using DevExpress.XtraRichEdit.Services;

namespace DevExpress.XtraRichEdit.Demos {
    public class CustomsRichEditCommandFactoryService : IRichEditCommandFactoryService {
        readonly IRichEditCommandFactoryService _service;
        readonly RichEditControl _control;
        readonly ButtonEdit _searchTextBox;

        public CustomsRichEditCommandFactoryService(RichEditControl richEditControl, IRichEditCommandFactoryService richEditCommandFactoryService, ButtonEdit searchTextBox) {
            _control = richEditControl;
            _service = richEditCommandFactoryService;
            _searchTextBox = searchTextBox;
        }

        RichEditCommand IRichEditCommandFactoryService.CreateCommand(RichEditCommandId id) {
            if(id.Equals(RichEditCommandId.Find))
                return new CustomFindCommand(_control, _searchTextBox);
            return _service.CreateCommand(id);
        }
    }
    public class CustomFindCommand : FindCommand {
        readonly ButtonEdit _searchTextBox;
        readonly char[] _separators = new char[] { Characters.ParagraphMark, Characters.PageBreak, Characters.TabMark };

        public CustomFindCommand(IRichEditControl richEditControl, ButtonEdit searchTextBox)
            : base(richEditControl) {
            _searchTextBox = searchTextBox;
        }

        protected override void ShowForm(string searchString) {
            _searchTextBox.Focus();
            string selectedText = Control.Document.GetText(Control.Document.Selection);
            if(string.IsNullOrEmpty(selectedText) || selectedText.IndexOfAny(_separators) >= 0)
                return;
            _searchTextBox.Text = selectedText.Trim();
        }
    }
}

File: WinForms/VB/RichEditMainDemo/Modules/CustomDraw/CustomsRichEditCommandFactoryService.vb
Imports DevExpress.Office
Imports DevExpress.XtraEditors
Imports DevExpress.XtraRichEdit.Commands
Imports DevExpress.XtraRichEdit.Services

Namespace DevExpress.XtraRichEdit.Demos

    Public Class CustomsRichEditCommandFactoryService
        Implements IRichEditCommandFactoryService

        Private ReadOnly _service As IRichEditCommandFactoryService

        Private ReadOnly _control As RichEditControl

        Private ReadOnly _searchTextBox As ButtonEdit

        Public Sub New(ByVal richEditControl As RichEditControl, ByVal richEditCommandFactoryService As IRichEditCommandFactoryService, ByVal searchTextBox As ButtonEdit)
            _control = richEditControl
            _service = richEditCommandFactoryService
            _searchTextBox = searchTextBox
        End Sub

        Private Function CreateCommand(ByVal id As RichEditCommandId) As RichEditCommand Implements IRichEditCommandFactoryService.CreateCommand
            If id.Equals(RichEditCommandId.Find) Then Return New CustomFindCommand(_control, _searchTextBox)
            Return _service.CreateCommand(id)
        End Function
    End Class

    Public Class CustomFindCommand
        Inherits FindCommand

        Private ReadOnly _searchTextBox As ButtonEdit

        Private ReadOnly _separators As Char() = New Char() {Characters.ParagraphMark, Characters.PageBreak, Characters.TabMark}

        Public Sub New(ByVal richEditControl As IRichEditControl, ByVal searchTextBox As ButtonEdit)
            MyBase.New(richEditControl)
            _searchTextBox = searchTextBox
        End Sub

        Protected Overrides Sub ShowForm(ByVal searchString As String)
            _searchTextBox.Focus()
            Dim selectedText As String = Control.Document.GetText(Control.Document.Selection)
            If String.IsNullOrEmpty(selectedText) OrElse selectedText.IndexOfAny(_separators) >= 0 Then Return
            _searchTextBox.Text = selectedText.Trim()
        End Sub
    End Class
End Namespace
